libraries/botbuilder-lg/src/evaluationOptions.ts

```diff
@@ -87,10 +87,7 @@ export class EvaluationOptions {
                                     this.strictMode = true;
                                 }
                             } else if (key.toLowerCase() === this.replaceNullKey.toLowerCase()) {
-                                // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                                this.nullSubstitution = (path) =>
-                                    // eslint-disable-next-line security/detect-eval-with-expression
-                                    eval('`' + value.replace(this.nullKeyReplaceStrRegex, '${path}') + '`'); // CodeQL [SM04509] Eval on content that is from a trusted source
+                                this.nullSubstitution = (path) => value.replace(this.nullKeyReplaceStrRegex, `${path}`);
```

Comment:
I think we could do this, i dont know if TS will complain about path not being a string or something.

```suggestion
                                this.nullSubstitution = (path) => value.replace(this.nullKeyReplaceStrRegex, path);
```

Reply:
Not sure if it's possible for path to be a function, but in that case, not using interpolation opens the door to XSS. This is due to the fact that `replace` supports an [overload that accepts a function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_the_replacement).
Say, for example:

```javascript
const value = "some ${path} thing"
const path = function () { window.alert('injected')};
value.replace(nullKeyReplaceStrRegex, `${path}`); // returns "some function () { window.alert('injected')} thing"
value.replace(nullKeyReplaceStrRegex, path); // Opens an alert window in your browser with the text "injected"
```
So, in principle interpolating path is, at least, safer.

Reply:
Cool thing to know, i didnt know string.replace works like that, just adding an image that proves it, thanks!.
![image](https://github.com/southworks/botbuilder-js/assets/62260472/6bb8f7dc-7a21-4cf5-9400-3d01ca01f61f)